src/app/labeling/qgsmaptoolrotatelabel.cpp

```diff
@@ -109,19 +109,9 @@ void QgsMapToolRotateLabel::canvasPressEvent( QgsMapMouseEvent *e )
     if ( !mCurrentLabel.valid )
       return;
 
-    // only rotate non-pinned OverPoint placements until other placements are supported in pal::Feature
```

Comment:
This is still the case, isn't it?

Reply:
No, but it was not the case even before, the comment don't say the same as the implementation. But maybe I missed something important, why should not be possible to rotate`OverPoint` labels?

Reply:
The code in the pal library for label placement totally ignores it -- it's just not implemented there yet

Reply:
See https://github.com/qgis/QGIS/blob/9ce295ebbd99719e856839a9c2b3c33c921d351f/src/core/labeling/qgspallabeling.cpp#L2464

Reply:
The `else` statement of line 2460 is not executed in normal cases. Only in (probably) rare cases when the X/Y expressions are defined but the returned values are not convertible to float.
If I change the rotation expression of an over point label (like by polygon mode "Offset from Centroid") it get rotated as expected.

Reply:
I agree with @domi4484, the whole block referenced above looks suspicious.
It will only be reached if datadefined x or y are enabled but invalid. In which case it also doesn't make sense to use the values `xPos` and `yPos` as done on line 2462.

https://github.com/qgis/QGIS/blob/9ce295ebbd99719e856839a9c2b3c33c921d351f/src/core/labeling/qgspallabeling.cpp#L2460-L2469

I would vote to merge this as is unless someone can come up with a good explanation of what's going on here (in which case I'm keen to hear it)